src/structs/lepton_decoder.rs

```diff
@@ -456,13 +487,14 @@ fn decode_one_edge<R: Read, const ALL_PRESENT: bool, const HORIZONTAL: bool>(
             num_non_zeros_edge -= 1;
         }
 
-        context.here_mut(image_data).set_coefficient(
+        block_context.here_mut(image_data).set_coefficient(
```

Comment:
could we pull this out of the loop, or is the borrow checker going to yell?

Reply:
you're right, I already cloned before the loop. Found that cloning here helps to pipeline since its easier for the compiler to understand that the calculations are all independent.